src/curve.rs

```diff
@@ -56,67 +61,292 @@ pub trait CT:
 
     /// Multiply `self` by a [`Scalar`].
     // TODO: name.
-    fn scale2(&self, point: &Scalar) -> Self;
+    fn scale2(&self, point: &Self::Scalar) -> Self;
 
     /// Compute the x-projection of the point.
-    fn x_projection(&self) -> Result<Scalar>;
+    fn x_projection(&self) -> Result<Self::Scalar>;
 
     /// Serialize the point as an affine-encoded byte array.
     fn to_bytes(self) -> Vec<u8>;
 
     /// Deserialize a point from an affine-encoded byte array.
     fn try_from_bytes(bytes: &[u8]) -> Result<Self>;
+
+    /// Convert BigNumber to Scalar.
+    fn bn_to_scalar(bn: &BigNumber) -> Result<Self::Scalar>;
+    
+    /// Multiply `self` by a [`BigNumber`] point, which is first converted to
+    /// the subjacent [`Scalar`] field (taken mod `q`, where `q` is the
+    /// order of the curve).
+    fn multiply_by_bignum(&self, point: &BigNumber) -> Result<Self>;
+    
+    fn multiply_by_scalar(&self, point: &Self::Scalar) -> Self;
+    
+    // Convert from Scalar to BigNumber
+   fn scalar_to_bn(x: &Self::Scalar) -> BigNumber;
 }
 
 /// Scalar trait.
-pub trait ST {}
+pub trait ST: 
+    Sync + Send + Clone + Copy + Debug + PartialEq + PartialOrd + Eq + Zeroize
+    + Serialize
+    + for<'de> Deserialize<'de>
+    + Add<Output = Self>
+    + AsRef<Self>
+    + Into<FieldBytes>
+{
+    /// Return the zero scalar.
+    fn zero() -> Self;
+
+    /// Return the one scalar.
+    fn one() -> Self;
+
+    /// Convert a u128 to a scalar.
+    fn convert_from_u128(x: u128) -> Self;
+
+    /// Add two scalars.
+    fn add(&self, other: &Self) -> Self;
+
+    /// Addition operator such that we can use += syntax.
+    fn add_assign(&mut self, other: Self) {
+        dbg!("add_assign");
+        *self = self.add(other);
+    }
+
+    /// Sub two scalars.
+    fn sub(&self, other: &Self) -> Self;
+
+    /// Negate
+    fn negate(&self) -> Self;
+
+    /// Multiply two scalars.
+    fn mul(&self, other: &Self) -> Self;
+
+    /// Implement the mul operator such that we can use *= syntax.
```

Comment:
Are these methods actually used? The actual syntax `*` and `*=` work with the traits `Mul` and `MulAssign` which can be added as requirements (`trait ST: Mul + …`).